cmd/kueuectl/app/list/list_workload_printer.go

```diff
@@ -0,0 +1,192 @@
+	return &listWorkloadResources{
+		localQueues:      make(map[string]*v1beta1.LocalQueue),
+		pendingWorkloads: make(map[string]*v1alpha1.PendingWorkload),
+		apiResourceLists: make(map[string]*metav1.APIResourceList),
+	}
+}
+
+type listWorkloadPrinter struct {
+	printOptions printers.PrintOptions
+	resources    *listWorkloadResources
+}
+
+var _ printers.ResourcePrinter = (*listWorkloadPrinter)(nil)
+
+func (p *listWorkloadPrinter) PrintObj(obj runtime.Object, out io.Writer) error {
+	printer := printers.NewTablePrinter(p.printOptions)
+
+	list, ok := obj.(*v1beta1.WorkloadList)
+	if !ok {
+		return fmt.Errorf("invalid object type")
+	}
+
+	table := &metav1.Table{
+		ColumnDefinitions: []metav1.TableColumnDefinition{
+			{Name: "Name", Type: "string", Format: "name"},
+			{Name: "CRD Type", Type: "string"},
+			{Name: "CRD Name", Type: "string"},
+			{Name: "LocalQueue", Type: "string"},
+			{Name: "ClusterQueue", Type: "string"},
+			{Name: "Status", Type: "bool"},
+			{Name: "Position in Queue", Type: "string"},
+			{Name: "Age", Type: "string"},
+		},
+		Rows: p.printWorkloadList(list),
+	}
+
+	return printer.PrintObj(table, out)
+}
+
+func (p *listWorkloadPrinter) WithNamespace() *listWorkloadPrinter {
+	p.printOptions.WithNamespace = true
+	return p
+}
+
+func (p *listWorkloadPrinter) WithResources(r *listWorkloadResources) *listWorkloadPrinter {
+	if r == nil {
+		r = newListWorkloadResources()
+	}
+	p.resources = r
+	return p
+}
+
+func newWorkloadTablePrinter() *listWorkloadPrinter {
+	return &listWorkloadPrinter{
+		resources: newListWorkloadResources(),
+	}
+}
+
+func (p *listWorkloadPrinter) printWorkloadList(list *v1beta1.WorkloadList) []metav1.TableRow {
+	rows := make([]metav1.TableRow, len(list.Items))
+	for index := range list.Items {
+		rows[index] = p.printWorkload(&list.Items[index])
+	}
+	return rows
+}
+
+func (p *listWorkloadPrinter) printWorkload(wl *v1beta1.Workload) metav1.TableRow {
+	row := metav1.TableRow{
+		Object: runtime.RawExtension{Object: wl},
+	}
+
+	var clusterQueueName string
+	if wl.Status.Admission != nil && len(wl.Status.Admission.ClusterQueue) > 0 {
+		clusterQueueName = string(wl.Status.Admission.ClusterQueue)
+	} else if lq := p.resources.localQueues[localQueueKeyForWorkload(wl)]; lq != nil {
+		clusterQueueName = string(lq.Spec.ClusterQueue)
+	}
+
+	var positionInQueue string
+	if pendingWorkload, ok := p.resources.pendingWorkloads[workload.Key(wl)]; ok {
+		positionInQueue = fmt.Sprintf("%d", pendingWorkload.PositionInLocalQueue)
+	}
+
+	row.Cells = []any{
+		wl.Name,
+		strings.Join(p.crdTypes(wl), ", "),
+		strings.Join(p.crdNames(wl), ", "),
+		wl.Spec.QueueName,
+		clusterQueueName,
+		strings.ToUpper(workload.Status(wl)),
+		positionInQueue,
+		duration.HumanDuration(time.Since(wl.CreationTimestamp.Time)),
+	}
+
+	return row
+}
+
+func (p *listWorkloadPrinter) crdTypes(wl *v1beta1.Workload) []string {
+	crdTypes := make(map[string]struct{}, len(wl.ObjectMeta.OwnerReferences))
```

Comment:
```suggestion
	crdTypes := sets.New[string]()
```

I would recommend using sets library to avoid maintaining such that map.